client/src/modules/install/install.js

```diff
@@ -0,0 +1,51 @@
+angular.module('bhima.controllers')
+  .controller('InstallApplicationController', InstallApplicationController);
+
+// dependencies injection
+InstallApplicationController.$inject = ['InstallService', '$state', 'NotifyService'];
+
+// controller definition
+function InstallApplicationController(InstallService, $state, Notify) {
+  var vm = this;
+
+  vm.setup = { enterprise : {}, project : {}, user : {} };
+
+  // expose methods to the view
+  vm.submit = submit;
+
+  InstallService.checkBasicInstallExist()
```

Comment:
This is pretty cool!

It looks like what this code will do is basically this:
navigate to /install -> load InstallApplicationController -> fire HTTP GET /install -> if already exist -> $state.go('login').

You could probably remove the step that loads the install controller if the `alreadyExist` flag is false in `onEnter` part of the `$state` change.  Something like this:
```js
onEnter: ['InstallService', '$state', function (InstallService, $state) {
  return InstallService.checkBasicInstallExist()
     .then(function (isInstalled) { 
       if (isInstalled) { $state.go('login'); }
     });
}]
```

Trying to catch the `isInstalled` flag before navigating to the route should make sure that there isn't a flash of content before re-routing.